verification/tests/solidity_multiple.rs

```diff
@@ -0,0 +1,398 @@
+        let test_input = TestInput::new("ContractFromFactory", "v0.4.26+commit.4563c3fc")
+            .with_optimization_runs(200)
+            .has_constructor_args();
+        test_success(contract_dir, test_input).await;
+    }
+}
+
+mod basic_error_tests {
+    use super::*;
+
+    #[actix_rt::test]
+    async fn returns_failure_when_bytecode_does_not_match() {
+        let contract_dir = "simple_storage";
+        let test_input = TestInput::new("SimpleStorage", "v0.4.24+commit.e67f0147")
+            .with_source_code("pragma solidity ^0.4.24; contract SimpleStorage {}".to_string());
+        test_failure(
+            contract_dir,
+            test_input,
+            "No contract could be verified with provided data",
+        )
+        .await;
+    }
+
+    #[actix_rt::test]
+    async fn returns_failure_with_compilation_problems() {
+        let contract_dir = "simple_storage";
+        let test_input = TestInput::new("SimpleStorage", "v0.4.24+commit.e67f0147")
+            .with_source_code("pragma solidity ^0.4.24; contract SimpleStorage { ".to_string());
+        test_failure(contract_dir, test_input, "ParserError").await;
+    }
+}
+
+mod regression_tests {
+    use super::*;
+
+    #[actix_rt::test]
+    async fn issue_5114() {
+        let contract_dir = "issue_5114";
+        let test_input = TestInput::new("TransparentUpgradeableProxy", "v0.8.2+commit.661d1103")
+            .with_optimization_runs(200)
+            .has_constructor_args();
+        test_success(contract_dir, test_input).await;
+    }
+
+    #[actix_rt::test]
+    async fn issue_5127() {
+        let contract_dir = "issue_5127";
+        let test_input = TestInput::new("YESToken", "v0.8.11+commit.d7f03943")
+            .with_optimization_runs(200)
+            .has_constructor_args();
+        test_success(contract_dir, test_input).await;
+    }
+
+    #[actix_rt::test]
+    async fn issue_3082() {
+        let contract_dir = "issue_3082";
+        let test_input = TestInput::new("Distribution", "v0.5.10+commit.5a6ea5b1")
+            .with_optimization_runs(200)
+            .has_constructor_args();
+        test_success(contract_dir, test_input).await;
+    }
+
+    #[actix_rt::test]
+    async fn another_failed_constructor_args_verification() {
+        let contract_dir = "issue_with_constructor_args";
+        let test_input = TestInput::new("ERC1967Proxy", "v0.8.2+commit.661d1103")
+            .with_optimization_runs(200)
+            .has_constructor_args();
+        test_success(contract_dir, test_input).await;
+    }
+
+    #[actix_rt::test]
+    async fn issue_4758() {
+        let contract_dir = "issue_4758";
+        let test_input = TestInput::new("CS3_1OnChainShop", "v0.8.4+commit.c7e474f2")
+            .with_optimization_runs(200)
+            .has_constructor_args();
+        test_success(contract_dir, test_input).await;
+    }
+
+    // runs only for linux, as other compiler lists do not have nightly builds
```

Comment:
right now nightly builds will be available only in s3, for which other developers don't have access

I have an idea to push all the nightly builds, which are used in tests in solc-bin releases, so we could use them without access to s3, we can discuss this